### ISSUE TITLE
CLARIFY: Steam DB is expired

### ISSUE THREAD

**Issue by fwdekker** [RimSort/RimSort#381]
When I start RimSort v1.0.7, I receive the message "Steam DB is expired! Consider updating!". However, it's not entirely clear to me what this means or how I should do this.

I have tried the following:
1. Go to "Settings" > "Databases".
2. Set "Community rules database" to "GitHub: https://github.com/RimSort/Community-Rules-Database" (the default).
3. Set "Steam Workshop database" to "GitHub: https://github.com/RimSort/Steam-Workshop-Database" (the default).
4. For both databases, I click "Download..." > "Clone new", and for both I receive the message "The configured repository was cloned!"
5. Press "OK" to close the settings.

If I then close and reopen RimSort, I still receive the aforementioned message about Steam DB being expired. I have tried looking at the "DB Builder" settings, but I have no idea what any of those settings mean, nor about how to configure them, nor about _why_ I should configure them, nor about whether those are even the correct settings to look at.

Concretely, I think the RimSort UX is confusing in the following ways, and should be clarified within the UI:
1. The error message when I start RimSort tells me there is an error, but does not tell me how to fix it.
2. The error message talks about a "Steam DB", but the term "Steam DB" is not used anywhere in the settings. The settings have a tab "Databases", but end users may not know that "DB" means "database". Furthermore, under that tab, there are two databases, but it is not clear which (if any) is the "Steam DB". Finally, the tab "DB Builder" may also be related, but it is not clear how this relates to the "Steam DB".
3. The [user guide](https://github.com/RimSort/RimSort/wiki/User-Guide) mentions a "SteamDB". However, the guide is somewhat outdated (in terms of screenshots/recordings), and does not explain what to do when the aforementioned error is seen.

I'm not opening this issue to complain. I truly, sincerely appreciate the fact that RimSort is free, gratis, and open source. My reason for opening this issue is to show concretely how the user experience can be improved in terms of clarity of terminology.

**Comment by AerosolNinja** [comment]
I too am receiving the Steam DB metadate expired error message and cannot determine a fix nor am I clear on what the issue actually means?

![image](https://github.com/RimSort/RimSort/assets/55816747/7d4b4562-2117-402c-9845-b0d1fd3b13af)

**Comment by hk21702** [comment]
So this should probably be clarified in the app, but essentially, what this warning means is that the date of generation for the Steam DB that you are currently using is older than the Database expiry that is set in `Settings > DB Builder > Database expiry`. 

Really, it's harmless and if you want the error to go away, you should be able to just bump that number up. That or, update the database, either by generating it yourself, or getting it from the community repository, assuming that repo has an updated version. At the time of writing, I don't think the repo has been updated in over a week, which is the default expiry. So, your options for now are to just ignore the message, or bump up the expiry, or update the database yourself.

**Comment by FWDekker** [comment]
Thank you for the explanation :-)

My question then is: Since I didn't build this database myself originally, where did it come from? I assume it was either included in the installation, or it's fetched automatically from somewhere. And is this database supposed to be a background thing or not? Because currently it's somewhere in between: Initially the end user doesn't even have to know it exists, but after a few days the end user is expected to be able to update the database.

Would it be possible to make the Steam DB update similar to the other two databases? That is, allow fetching an updated version over the Internet? Either way, I don't think an apparently harmless situation should result in an error/warning dialog.

**Comment by hk21702** [comment]
It is fetched from the community database repository. It works the same as the community rules repository. It's just that the community DB hasn't been updated in a little while. 

See the wiki for more info on how the DB works, what it's used for, and how to build your own.

**Comment by LionelColaso** [comment]
https://github.com/RimSort/RimSort/wiki/User-Guide#external-metadata

![image](https://github.com/RimSort/RimSort/assets/2488703/0a225c71-6cc4-4c3c-ad95-6d62cebcb17e)

**Comment by FWDekker** [comment]
I had looked through that page before (as indicated in my issue description), but still found the link between the error message and the settings to be unclear. Your messages imply that "Steam Workshop Database" = "Steam DB", but the difference in terminology made me suspect that there may have been a third, unrelated database called "Steam DB".

Therefore, suggestions (1) and (2) from my original issue description about locality of information and consistency of terminology still hold, point (3) obviously doesn't make much sense anymore, but additionally I think that (4) there should not be a warning if the community database has not been updated.

**Comment by LionelColaso** [comment]
you would only get a warning for SteamDB being old there is no warning for CommunityDB

**Comment by hk21702** [comment]
There probably can be some function to check that if you are selected to use the community steam workshop DB repository that the warning only shows if the time expired and there is an update in that repository. However, the warning itself isn't invalid as you can build a database yourself, or change the trigger for it.

I do agree that the message should likely tell you how to fix it and that some clarifications can be done on what things mean. Most likely, we won't expand DB to Database for spacing reasons, but I've been considering adding tooltips throughout and this could be a good example of where to add it. However, it is currently a low-priority thing since there's a lot of hard breaking issues currently

**Comment by FWDekker** [comment]
@LionelColaso Oops, you're correct, I meant the other one ^^'

@hk21702 Thank you for considering :-) In terms of consistency, I think the only thing that really needs changing is that the alert says "Steam DB" instead of "Steam Workshop database", like in the settings. And I completely understand that this is low priority, no issue there :-)

**Comment by Raizanad** [comment]
Also, it shouldn't be a popup but something else. It is annoying and freezes the app from doing other things until closed.

**Comment by LionelColaso** [comment]
Fixed in latest Version
